pkg/chartutil/values.go

```diff
@@ -281,6 +281,13 @@ func coalesceValues(c *chart.Chart, v map[string]interface{}) (map[string]interf
 		if _, ok := v[key]; !ok {
 			// If the key is not in v, copy it from nv.
 			v[key] = val
+		} else if ok && v[key] == nil {
```

Comment:
- [ ] Here, we could have avoided a second map lookup by binding the mapped value to a variable at line 281 above:
```go
if value, ok := v[key]; ! ok {
  // ...
} else if ok && value == nil {
  // When the YAML ...
}
```
I'd consider then turning the conditional around:
```go
if value, ok := v[key]; ok {
  if value == nil {
    // When the YAML ...
  } else if dest, ok := value.(map[string]interface{}); ok {
    // If v[key] is a table ...
  }
} else {
  // If the key is not in v ...
}
```